src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,107 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { getPostDetails } from '../../api/posts';
+import { deletePostComment, getPostComments, postNewComment } from '../../api/comments';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+type Props = {
+  postId: number;
+};
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+export const PostDetails: React.FC<Props> = ({ postId }) => {
+  const [post, setPost] = useState<Post>({} as Post);
+  const [comments, setComments] = useState<PostComment[]>([]);
+  const [hideComments, setHideComments] = useState(false);
+  const [showComments, setShowComments] = useState(true);
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+  useEffect(() => {
+    const fetchData = async () => {
+      const [postFromServer, commentsFromServer] = await Promise.all([
+        getPostDetails(postId),
+        getPostComments(postId),
+      ]);
 
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
+      setPost(postFromServer[0]);
+      setComments(commentsFromServer);
+    };
+
+    fetchData();
+  }, [postId]);
+
+  const deleteComment = async (commentId: number) => {
+    await deletePostComment(commentId);
+
+    setComments(currentComments => currentComments.filter(
+      comment => comment.id !== commentId,
+    ));
+  };
+
+  const addComment = async (comment: PostComment) => {
+    const newComment = await postNewComment(comment);
+
+    setComments(currentComments => [...currentComments, newComment]);
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        {post !== null && <p>{post.body}</p>}
+      </section>
+
+      <section className="PostDetails__comments">
+        {!hideComments ? (
           <button
             type="button"
-            className="PostDetails__remove-button button"
+            className="button PostDetails__button"
+            onClick={() => {
+              setHideComments(true);
+              setShowComments(false);
+            }}
+            disabled={comments.length < 1}
           >
-            X
+            Hide comments
```

Comment:
let it be just one button, just change the text inside depending on the boolean of isCommentsHidden